src/com/goide/inspections/GoDeferGoInspection.java

```diff
@@ -35,17 +39,40 @@ protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNul
       @Override
       public void visitDeferStatement(@NotNull GoDeferStatement o) {
         super.visitDeferStatement(o);
-        checkExpression(o.getExpression(), "defer");
+        checkExpression(o, o.getExpression(), "defer");
       }
 
       @Override
       public void visitGoStatement(@NotNull GoGoStatement o) {
         super.visitGoStatement(o);
-        checkExpression(o.getExpression(), "go");
+        checkExpression(o, o.getExpression(), "go");
       }
 
-      private void checkExpression(@Nullable GoExpression o, String who) {
-        if (o == null || o instanceof GoCallExpr || o instanceof GoBuiltinCallExpr) return;
+      private void checkExpression(@NotNull GoStatement statement, @Nullable GoExpression o, String who) {
+        if (o == null || o instanceof GoBuiltinCallExpr) return;
+        if (o instanceof GoCallExpr) {
+          if (GoPsiImplUtil.isPanic((GoCallExpr)o)) {
```

Comment:
actually, the new control flow is strange a little. Firstly we have two different methods for `go` and `defer` statement and we join them in a single `checkExpression` method. And then we separate them again by conditions like `if ("go".equals(who))`, so why did we join them in the first place?